### PR TITLE
Prevent the Action from failing with strict mode disabled

### PR DESCRIPTION
**Pull request by lucperkins**

### CODE REVIEW

**Review by colemickens — APPROVED** [review]

File: src/index.ts [hunk 1]
```diff
@@ -106,19 +108,23 @@ class MagicNixCacheAction extends DetSysAction {
...
 
   async post(): Promise<void> {
-    if (this.noopMode) {
-      actionsCore.debug(TEXT_NOOP);
-      return;
-    }
+    if (!this.strictMode && this.errorInMain) {
+      this.exitWithWarning(`skipping post phase due to error in main phase`);
+    } else {
+      if (this.noopMode) {
```
Comment (cole-h) [inline src/index.ts:113]: Nit: why does this need to be in the else branch? I feel like it made more sense to check "if no-op, then return immediately" than this

File: src/index.ts [hunk 2]
```diff
@@ -350,11 +374,16 @@ class MagicNixCacheAction extends DetSysAction {
...
     actionsCore.debug(`killing`);
+
     try {
       process.kill(pid, "SIGTERM");
-    } catch (e) {
+    } catch (e: unknown) {
       if (typeof e === "object" && e && "code" in e && e.code !== "ESRCH") {
-        throw e;
+        if (this.strictMode) {
+          throw e;
```
Comment (cole-h) [inline src/index.ts:383]: There seems to be a lot of this "if strict then ... else ..." -- would it make sense to abstract that into a function?

File: src/index.ts [hunk 3]
```diff
@@ -106,19 +107,25 @@ class MagicNixCacheAction extends DetSysAction {
...
-    if (this.noopMode) {
-      actionsCore.debug(TEXT_NOOP);
-      return;
-    }
+    // If strict mode is off and there was an error in main, such as the daemon not starting,
+    // then the post phase is skipped with a warning.
+    if (!this.strictMode && this.mainError) {
+      this.exitWithWarning(this.mainError);
+    } else {
+      if (this.noopMode) {
```
Comment (colemickens) [inline src/index.ts]: what is noopMode? Should that conditional be pulled out and be a first fast exit? Or can mainError happen even in noopMode?

Comment (lucperkins) [reply src/index.ts]: Ah, you're right. No-op mode pretty much does what I'm trying to accomplished with the "error on main" logic. I'll remove it.

Comment (lucperkins) [reply src/index.ts]: Actually, strike that: the difference is that "no-op mode" means that the daemon is already running and that it doesn't need to be started again. So having both "branches" makes sense here. But maybe I can adjust the naming to make it more intuitive.

File: src/index.ts (hunk 2 above)
Comment (lucperkins) [reply src/index.ts:387]: The usages are subtly different enough that I think it's best to keep it like it is

**Pull request merged**